variables.tf

```diff
@@ -67,3 +67,11 @@ variable "labels" {
   type        = map(string)
   default     = {}
 }
+variable "lifecycle_rules" {
+  description = "List of lifecycle rules to configure. Format is the same as described in provider documentation https://www.terraform.io/docs/providers/google/r/storage_bucket.html#lifecycle_rule except condition.matches_storage_class should be a comma delimited string."
```

Comment:
Actually this text maybe misleading because we are not supporting ALL arguments of the provider as mentioning here - we are supporting only 6 of its many arguments. Maybe better to clarify in the description which arguments are being supported

Reply:
I don't think there is any use case to use other arguments, if we need it later we add it. 
the description has nothing to do with how many arguments we support. it should be in the documentation and how we use it.

Reply:
```suggestion
  description = "List of lifecycle rules to configure. Defines action.type, action.storage_class, condition.age, condition.created_before, condition.with_state, condition.matches_storage_class, condition.num_newer_versions. Format is same as described in provider documentation https://www.terraform.io/docs/providers/google/r/storage_bucket.html#lifecycle_rule. Except condition.matches_storage_class should be a comma delimited string."
```